contracts/test/PDPService.t.sol

```diff
@@ -62,11 +61,116 @@ contract PDPServiceProofSetCreateDeleteTest is Test {
+            1, 2, 1, 3, 1, 2, 1, 4, 1, 2, 1, 3, 1, 2, 1, 5, 1, 2, 1, 3, 1, 2, 1, 4, 1, 2, 1, 3, 1, 2, 1, 6, 
+            1, 2, 1, 3, 1, 2, 1, 4, 1
+        ];
+        for (uint32 i = 0; i < 105; i++) {
+            assertEq(oeisA001511[i], pdpService.testHeightFromIndex(i) + 1, "Heights from index 0 to 104 should match OEIS A001511");
+        }
+    }
+}
+
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "../src/PDPService.sol";
+
+contract SumTreeAddTest is Test {
+    PDPService pdpService;
+    uint256 testSetId;
+
+    function setUp() public {
+        pdpService = new PDPService(100); // Assuming 100 as challengeFinality
+        testSetId = pdpService.createProofSet();
+    }
+
+    function testMultiAdd() public {
+        uint256[] memory sizes = new uint256[](8);
+        sizes[0] = 5;
+        sizes[1] = 5;
+        sizes[2] = 5;
+        sizes[3] = 5;
+        sizes[4] = 5;
+        sizes[5] = 5;
+        sizes[6] = 5;
+        sizes[7] = 5;
+
+        PDPService.RootData[] memory rootDataArray = new PDPService.RootData[](8);
+
+        for (uint256 i = 0; i < sizes.length; i++) {
+            PDPService.Cid memory testCid = PDPService.Cid(abi.encodePacked("test", i));
+            rootDataArray[i] = PDPService.RootData(testCid, sizes[i] * pdpService.CHUNK_SIZE());
+        }
+        pdpService.addRoot(testSetId, rootDataArray);
+        assertEq(pdpService.getProofSetSize(testSetId), 40, "Incorrect final proof set size");
+    }
+
+    function testSumTreeAdd() public {
+        uint256[] memory sizes = new uint256[](8);
+        sizes[0] = 200;
+        sizes[1] = 100;
+        sizes[2] = 0;
+        sizes[3] = 30;
+        sizes[4] = 50;
+        sizes[5] = 0;
+        sizes[6] = 400;
+        sizes[7] = 40;
+
+        uint256[] memory expectedSumTreeSizes = new uint256[](8);
+        expectedSumTreeSizes[0] = 200;
+        expectedSumTreeSizes[1] = 300;
+        expectedSumTreeSizes[2] = 0;
+        expectedSumTreeSizes[3] = 330;
+        expectedSumTreeSizes[4] = 50;
+        expectedSumTreeSizes[5] = 50;
+        expectedSumTreeSizes[6] = 400;
+        expectedSumTreeSizes[7] = 820;
+
+        for (uint256 i = 0; i < sizes.length; i++) {
+            PDPService.Cid memory testCid = PDPService.Cid(abi.encodePacked("test", i));
+            PDPService.RootData[] memory rootDataArray = new PDPService.RootData[](1);
+            rootDataArray[0] = PDPService.RootData(testCid, sizes[i] * pdpService.CHUNK_SIZE());
+            pdpService.addRoot(testSetId, rootDataArray);
+
+            // Assert that the root was added
+            assertEq(pdpService.getRootCid(testSetId, i).data, testCid.data, "Root not added correctly");
+
+            // Assert that the sum tree size is correct
+            assertEq(pdpService.getSumTreeSize(testSetId, i), expectedSumTreeSizes[i], "Incorrect sum tree size");
```

Comment:
I can't comment on where this function is defined now, but I don't think the API should be exposing sum trees. This data structure is an implementation detail of a logical contract of the PDP service. (It might still be handy to make available internally for testing, though).

However, it might make sense to expose a method like getRootAtOffset, which takes an offset into the logical array of the concatenation of all the roots and returns the corresponding root ID.

Reply:
solidity actually makes it easy to expose different functionality for testing so it was straightforward to remove the getter altogether